src/main/java/de/tum/in/www1/exerciseapp/service/ExerciseService.java

```diff
@@ -31,10 +32,13 @@
     private final UserService userService;
     private final ParticipationService participationService;
 
-    public ExerciseService(ExerciseRepository exerciseRepository, UserService userService, ParticipationService participationService) {
+    private Optional<ContinuousIntegrationService> continuousIntegrationService;
```

Comment:
please make it final as the other services above

Reply:
Done.